### DISCUSSION TITLE
Wiznet W5100S

### DISCUSSION THREAD

**Discussion by Chubls** [khoih-prog/EthernetWebServer#56] [General]
Hi All,

I'm quite new to this so i'm probably doing something wrong, or putting this in the wrong place.

I currently have a RP2040 rigged up to a Wiznet Ethernet Hat for the Pico (Uses the W5100S Chip). So ive set the CS pin to 17 and all the numbers tally up correctly.

I've run the "HelloServer" example, the code compiles but i get the following;

![image](https://user-images.githubusercontent.com/74865882/164262707-7e2549f1-2782-4089-9ddc-81f60647cfa5.png)

Any help with what i'm doing wrong? or does this library not support the S model?

Thanks

**Comment by khoih-prog** [comment]
I don't have that `Wiznet Ethernet Hat for the Pico` to test to know what's wrong, but from the connection, the 2 pins INTn and RSTn are connected directly to GPIO21 and GPIO20 and can cause issue to the firmware, if not pulled HIGH.

![Wiznet-Ethernet-Hat_RP2040_pinout](https://user-images.githubusercontent.com/57012152/164311201-2f84d0ff-5fde-4771-9eed-3f3881cadcb3.png)

I suggest that you add to the beginning of the code something like to see if it's OK

```
#define PIN_RSTN      20
#define PIN_INTN      21

pinMode(PIN_RSTN, OUTPUT);
pinMode(PIN_INTN, INPUT_PULLUP);
 
digitalWrite(PIN_RSTN, LOW);
delay(1000);
digitalWrite(PIN_RSTN, HIGH);
```

If not, I have no idea untiI have the board to test and investigate. You can also post an issue on Wiznet website to ask for support,

Good Luck.

**Reply by khoih-prog** [reply]
The data is wrong here [WIZnet Ethernet HAT](https://docs.wiznet.io/Product/Open-Source-Hardware/wiznet_ethernet_hat)

![Selection_143](https://user-images.githubusercontent.com/57012152/164533993-39103596-424b-4e81-92ff-186473d0798f.png)

Pin INTn is OUTPUT, not INPUT for W5100s
Pin RSTn is correctly INPUT  for W5100s

Any Luck so far?

**Comment by khoih-prog** [comment]
Hi,

I just bought and received one [**WIZnet Ethernet HAT**](https://docs.wiznet.io/Product/Open-Source-Hardware/wiznet_ethernet_hat) and have changed the [**Ethernet_Generic** library](https://github.com/khoih-prog/Ethernet_Generic) to provide preliminary support that `W5100S HAT` for both RP2040 `arduino-pico` and [ArduinoCore-mbed](https://github.com/arduino/ArduinoCore-mbed) cores.

Will release a new version with several days.

---

![Selection_145](https://user-images.githubusercontent.com/57012152/164574534-89594e79-9d8b-4938-8c45-77cc415be3b2.png)


```
Starting AdvancedWebServer on MBED RASPBERRY_PI_PICO with W5x00 using Ethernet_Generic Library on SPI0/SPI
EthernetWebServer v2.1.2
[EWS] =========== USE_ETHERNET_GENERIC ===========
[EWS] Default SPI pinout:
[EWS] MOSI: 19
[EWS] MISO: 16
[EWS] SCK: 18
[EWS] SS: 17
[EWS] =========================
[EWS] RPIPICO setCsPin: 17
[ETG] W5100 init, using SS_PIN_DEFAULT = 17 , new ss_pin =  10 , W5100Class::ss_pin =  17
[ETG] isW5100S: detect W5100S chip
[ETG] readMR (10) = 19
[ETG] readMR (18) = 27
[ETG] Version = 81
[ETG] Chip is W5100S
[ETG] W5100::init: W5100S, SSIZE = 4096
[EWS] =========================
[EWS] Currently Used SPI pinout:
[EWS] MOSI: 19
[EWS] MISO: 16
[EWS] SCK: 18
[EWS] SS: 17
[EWS] =========================
Using mac index = 16
Connected! IP address: 192.168.2.92
HTTP EthernetWebServer is @ IP : 192.168.2.92
.....[EWS] String Len = 0, extend to 2048
..... .......... .......... ...
```

---

![Selection_144](https://user-images.githubusercontent.com/57012152/164574624-bbc516e3-9f49-4a8a-aefc-7ffbd9233edd.png)

```
Starting AdvancedWebServer on RASPBERRY_PI_PICO with W5x00 using Ethernet_Generic Library on SPI0/SPI
EthernetWebServer v2.1.2
[EWS] =========== USE_ETHERNET_GENERIC ===========
[EWS] Default SPI pinout:
[EWS] MOSI: 19
[EWS] MISO: 16
[EWS] SCK: 18
[EWS] SS: 17
[EWS] =========================
[EWS] RPIPICO setCsPin: 17
[ETG] W5100 init, using SS_PIN_DEFAULT = 17 , new ss_pin =  10 , W5100Class::ss_pin =  17
[ETG] isW5100S: detect W5100S chip
[ETG] readMR (10) = 19
[ETG] readMR (18) = 27
[ETG] Version = 81
[ETG] Chip is W5100S
[ETG] W5100::init: W5100S, SSIZE = 4096
[EWS] =========================
[EWS] Currently Used SPI pinout:
[EWS] MOSI: 19
[EWS] MISO: 16
[EWS] SCK: 18
[EWS] SS: 17
[EWS] =========================
Using mac index = 17
Connected! IP address: 192.168.2.88
HTTP EthernetWebServer is @ IP : 192.168.2.88
.[EWS] String Len = 0, extend to 2048
.
```

**Comment by khoih-prog** [comment]
The [**Ethernet_Generic** releases v2.1.0](https://github.com/khoih-prog/Ethernet_Generic/releases/tag/v2.1.0) has been published to provide support to W5100S shield /module, such as [**WIZnet Ethernet HAT**](https://docs.wiznet.io/Product/Open-Source-Hardware/wiznet_ethernet_hat) and [**W5100S-EVB-Pico**](https://docs.wiznet.io/Product/iEthernet/W5100S/w5100s-evb-pico)

---

### Releases v2.1.0

1. Add support to WIZNet W5100S. Check [**Wiznet W5100S** #56](https://github.com/khoih-prog/EthernetWebServer/discussions/56)
